episodes/07-commandargs.md

```diff
@@ -0,0 +1,101 @@
+---
+title: "Using command-line arguments"
+teaching: 60
+exercises: 30
+---
+
+:::::::::::::::::::::::::::::::::::::: questions
+- "How do I use the same program for multiple use-cases?"
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+::::::::::::::::::::::::::::::::::::: objectives
+- "First objective."
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+From the last run of our code, we can see that 500 iterations is not enough to get to a _steady state_ (a
+state where the difference in temperature does not vary too much, i.e. `curdif`<`mindif`). Now, if we want to
+change the number of iterations we would need to modify `niter` in the code, and compile it again.  What if we
+want to change the number of rows and columns in our grid to have more precision, or if we want to see the
+evolution of the temperature at a different point (x,y)? The answer would be the same, modify the code and
+compile it again!
+
+No need to say that this would be very tedious and inefficient. A better scenario would be if we can pass the
+desired configuration values to our binary when it is called at the command line. The Chapel mechanism for
+this is to use **_config_** variables. When a variable is declared with the `config` keyword, in addition to
+`var` or `const`, like this:
+
+```chpl
+config const niter = 500;    //number of iterations
+```
+
+```bash
+>> chpl base_solution.chpl -o base_solution
+```
+
+it can be initialised with a specific value, when executing the code at the command line, using the syntax:
+
+```bash
+>> ./base_solution --niter=3000
+```
+
+```output
+The simulation will consider a matrix of 100 by 100 elements,
+it will run up to 3000 iterations, or until the largest difference
+in temperature between iterations is less than 0.0001.
+You are interested in the evolution of the temperature at the 
+position (1,100) of the matrix...
+
+and here we go...
+Temperature at iteration 0: 25.0
+Temperature at iteration 20: 2.0859
+Temperature at iteration 40: 1.42663
+...
+Temperature at iteration 2980: 0.793969
+Temperature at iteration 3000: 0.793947
+
+Final temperature at the desired position after 3000 iterations is: 0.793947
+The greatest difference in temperatures between the last two iterations was: 0.000350086
+```
+
+::::::::::::::::::::::::::::::::::::: challenge
+
+## Challenge 4: Can you do it?
+
+Make `n`, `x`, `y`, `mindif`, `rows` and `cols` configurable variables, and test the code simulating different
+configurations. What can you conclude about the performance of the code?
+
+:::::::::::::::::::::::: solution
```

Comment:
This may be a useful hint, but it is not a solution.

Reply:
I provided the solution to Challenge 4. Resolved through a commit.